vars/checkChangeInDevelopment.groovy

```diff
@@ -0,0 +1,89 @@
+import com.sap.piper.GitUtils
+import groovy.transform.Field
+
+import com.sap.piper.ConfigurationMerger
+import com.sap.piper.cm.ChangeManagement
+import com.sap.piper.cm.ChangeManagementException
+
+@Field def STEP_NAME = 'checkChangeInDevelopment'
+
+@Field Set parameterKeys = [
+    'changeDocumentId',
+    'cmClientOpts',
+    'credentialsId',
+    'endpoint',
+    'failIfStatusIsNotInDevelopment',
+    'git_from',
+    'git_to',
+    'git_label',
+    'git_format'
+  ]
+
+@Field Set stepConfigurationKeys = [
+    'changeDocumentId',
+    'cmClientOpts',
+    'credentialsId',
+    'endpoint',
+    'failIfStatusIsNotInDevelopment',
+    'git_from',
+    'git_to',
+    'git_label',
+    'git_format'
+  ]
+
+def call(parameters = [:]) {
+
+    handlePipelineStepErrors (stepName: STEP_NAME, stepParameters: parameters) {
+
+        prepareDefaultValues script: this
+
+        GitUtils gitUtils = parameters?.gitUtils ?: new GitUtils()
+
+        ChangeManagement cm = parameters?.cmUtils ?: new ChangeManagement(parameters.script, gitUtils)
+
+        Map configuration = ConfigurationMerger.merge(parameters.script, STEP_NAME,
+                                                      parameters, parameterKeys,
+                                                      stepConfigurationKeys)
+
+
+        def changeId
+
+        try {
+
+            changeId = cm.getChangeDocumentId(configuration)
+
+            if(! changeId?.trim()) {
+                throw new ChangeManagementException("ChangeId is null or empty.")
+            }
+        } catch(ChangeManagementException ex) {
+            throw new hudson.AbortException(ex.getMessage())
+        }
+
```

Comment:
We could add a log message like: Checking if changeId '$changeId' is in development.

Reply:
done